### ISSUE TITLE
Inconsistent datatypes for counter values and history

### ISSUE THREAD

**Issue by RasmusWesterlundh** [domoticz/domoticz#4955]
### Issue:
Both managed counters and counters can hold decimal values as the counter value.
But they cannot store decimal values in the history-log since the datatypes in the `Meter`-table in the database only accept `INTEGER` values.

Since values can be stored as decimals I suggest changing the datatype in the database to something that lets you store actual values to the history log.

### Suggestion:
Change the datatype of the `Value` and `Usage` in the `Meter`-table to `FLOAT`

**Comment by gizmocuz** [comment]
A counter value is a whole number... 1,2,3 and not 1,1.3,1.7
Sure current usage can be 3.5 Watt ... maybe start playing with out JSON api

**Comment by RasmusWesterlundh** [comment]
@gizmocuz I have been playing around with the API and I'm sorry sir but it does not matter.
You can for a fact not store the historical values of a counter since the counter can have decimal values and the historical (meter-table) cannot hold decimal values.
So the counter can have values that cannot be stored in the log - **this is an inconsistency and an issue**. Either the Counter should only allow Ints or the database should allow decimals.

I'm currently working on a PR on this but seeing your response to this I assume help is not wanted.

**Comment by gizmocuz** [comment]
I think you are missing the point of a counter .... 1,2.3 ... it just counts and should be a complete number
Consider your kWh meter at home ... it counts 10001, 10002 , 10003 ... not 10003.567 
What hardware are you using/creating that needs this precision ?

**Comment by RasmusWesterlundh** [comment]
@gizmocuz I have several Managed Counters for this use case, and I have had for years, only recently I got intrested in the history log for those counters.
One example is I buy my electricity from market spot prices, so I have one Managed Counter with the current spot price for electricity. This value can range anywhere from `-0.2999` to `2.4999` during the hours of the day. The current price is with 4 decimals precision.
I have other counters also related to gains, losses and percentages that also require at least 2 decimals precision. 

Managed counters currently support this. But the inconsistensy is in the database `Meters`-table for historical data for the Counter.
